xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxBXA.h

```diff
@@ -56,17 +56,19 @@ class CDVDDemuxBXA : public CDVDDemux
 
   bool Open(CDVDInputStream* pInput);
   void Dispose();
-  void Reset();
-  void Abort();
-  void Flush();
-  DemuxPacket* Read();
+
+  // implementation of CDVDDemux
+  virtual void Reset() override;
+  virtual void Abort() override;
+  virtual void Flush() override;
+  virtual DemuxPacket* Read() override;
   bool SeekTime(double time, bool backwards = false, double* startpts = NULL) override { return false; }
-  void SetSpeed(int iSpeed) {};
-  int GetStreamLength() { return (int)m_header.durationMs; }
+  virtual void SetSpeed(int iSpeed) override {}
```

Comment:
lifetime of input stream is not bound to lifetime of demuxer. SetSpeed was hacked into the code violating a basic OO principle: information hiding